### PR TITLE
Added fixes and improvements to phpdoc.

### PR DESCRIPTION
**Pull request by GhaziTriki**
## Type

- Enhancement

## Resolves the following issues

Some phpdoc are outdated, missing or wrong.

## Pull request description

Adds `@throws`, `@return` where missing and fix wrong variable names.

### CODE REVIEW

File: library/external/campaignmonitor.php [hunk 1]
```diff
@@ -125,13 +125,13 @@ class CampaignMonitor
...
 	 * Class constructor
 	 *
-	 * @return	void
 	 * @param	string $URL						The base URL of your CreateSend site. e.g. http://example.createsend.com/.
 	 * @param	string $username				The username you use to login to Campaign Monitor.
 	 * @param	string $password				The password you use to login to Campaign Monitor.
      * @param	int[optional] $timeOut			The default timeout
 	 * @param	string[optional] $clientId		The default client ID to use throughout the class.
 	 * @param	string[optional] $listId		The default list ID to use throughout the class.
+     * @throws CampaignMonitorException
```
Comment (WouterSioen) [inline library/external/campaignmonitor.php]: looks like the indentation of the timeOut parameter isn't correct too.

**Comment by WouterSioen** [comment]
👍

File: library/external/campaignmonitor.php [hunk 2]
```diff
@@ -122,17 +122,17 @@ class CampaignMonitor
...
 
-	/**
-	 * Class constructor
-	 *
-	 * @return	void
-	 * @param	string $URL						The base URL of your CreateSend site. e.g. http://example.createsend.com/.
-	 * @param	string $username				The username you use to login to Campaign Monitor.
-	 * @param	string $password				The password you use to login to Campaign Monitor.
+    /**
+     * Class constructor
```
Comment (carakas) [inline library/external/campaignmonitor.php]: this is a bit of a useless comment, could be removed but I don't think this is a blocker

File: src/Backend/Core/Engine/Navigation.php [hunk 3]
```diff
@@ -116,11 +116,20 @@ public function buildCache()
...
+    /**
+     * @param TwigTemplate $template
+     */
     public function parse(TwigTemplate $template)
     {
         $template->assign('navigation', $this->navigation);
     }
 
+    /**
+     * @param     $navigation
```
Comment (carakas) [inline src/Backend/Core/Engine/Navigation.php]: this one is still missing a type

File: src/Backend/Modules/Analytics/GoogleClient/ClientFactory.php [hunk 4]
```diff
@@ -21,6 +21,12 @@ class ClientFactory
      */
     private $cacheDir;
 
+    /**
+     * ClientFactory constructor.
```
Comment (carakas) [inline src/Backend/Modules/Analytics/GoogleClient/ClientFactory.php]: looks like a useless comment

File: src/Backend/Modules/Analytics/GoogleClient/Connector.php [hunk 5]
```diff
@@ -26,6 +26,13 @@
      */
     private $settings;
 
+    /**
+     * Connector constructor.
```
Comment (carakas) [inline src/Backend/Modules/Analytics/GoogleClient/Connector.php]: idem

File: src/Backend/Modules/Faq/DataFixtures/LoadFaqCategories.php [hunk 6]
```diff
@@ -4,6 +4,9 @@
 
 class LoadFaqCategories
 {
+    /**
+     * @param \SpoonDatabase $database
+     */
     public function load(\SpoonDatabase $database)
```
Comment (carakas) [inline src/Backend/Modules/Faq/DataFixtures/LoadFaqCategories.php]: maybe add a use statement for SpoonDatabase

File: src/Backend/Modules/Faq/DataFixtures/LoadFaqQuestions.php [hunk 7]
```diff
@@ -4,6 +4,9 @@
 
 class LoadFaqQuestions
 {
+    /**
+     * @param \SpoonDatabase $database
+     */
     public function load(\SpoonDatabase $database)
```
Comment (carakas) [inline src/Backend/Modules/Faq/DataFixtures/LoadFaqQuestions.php]: maybe add a use statement for SpoonDatabase

File: src/Backend/Modules/Profiles/Engine/Model.php [hunk 8]
```diff
@@ -631,9 +631,14 @@ public static function getUser($id)
      * Import CSV data
      *
      * @param array $data The array from the .csv file
-     * @param int[optional] $groupId Adding these profiles to a group
-     * @param bool[optional] $overwriteExisting If set to true, this will overwrite existing profiles
-     * @param return array('count' => array('exists' => 0, 'inserted' => 0));
+     * @param       int   [optional] $groupId Adding these profiles to a group
```
Comment (carakas) [inline src/Backend/Modules/Profiles/Engine/Model.php]: spacing looks off here

**Comment by carakas** [comment]
Some of the phpdoc is now aligned with each other, this might lead to otherwise unneeded added spaces in those rows when adding a parameter to the phpdoc.
I would remove the alignment spaces, but I don't see this as a blocker

also, is [optional] valid phpdoc?

**Comment by WouterSioen** [comment]
`[optional]` is no valid phpdoc

**Pull request merged**